### ISSUE TITLE
Error when mounting a collection

### ISSUE THREAD

**Issue by fxkopp** [statamic/cms#1712]
##  Bug Description
When I try to mount a collection to a Page it gives me a `Call to a member function handle() on null` error and the dialogue to select a Page is completely empty.

## How to Reproduce
Create a collection, go to 'Configure Collection', scroll down to 'Routing & URLs' and in section 'Mount' click on 'Link Existing Item'.

## Extra Detail
![image](https://user-images.githubusercontent.com/1269390/79842829-a9d94b80-83b9-11ea-9b07-eeebbf90bedd.png)

```
Call to a member function handle() on null
/vendor/statamic/cms/src/Fieldtypes/Relationship.php#79
```

```php
 return $items->map(function ($item) use ($data) {
            return [
                'id' => method_exists($item, 'id') ? $item->id() : $item->handle(),
                'title' => method_exists($item, 'title') ? $item->title() : $item->get('title'),
                'edit_url' => $item->editUrl(),
                'published' => $this->statusIcons ? $item->published() : null,
```

## Environment

**Statamic version:** 3.0.0.-beta-26

**PHP version:** 7.4.3

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by damcclean** [comment]
I have replicated this issue. It only happens when an entry has a relationship field (eg. an Entry fieldtype) and it points to an entry that no longer exists.

**Comment by jasonvarga** [comment]
Fixed in 7f06fc60
